### DISCUSSION TITLE
Snap on/after/during draw

### DISCUSSION THREAD

**Discussion by rjtwins** [Blazor-Diagrams/Blazor.Diagrams#354] [Q&A] [answered]
I'm drawing my diagram from information from an external source (including coordinates), is it possible to make the nodes snap the the nearest snapping point on/after drawing?

**Comment by zHaytam** [comment, marked as answer]
Currently, there is no out of the box to do that, however you can just re-set the position of all your nodes while while snapping it with the following snippet of code:
```cs
yourNode.SetPosition(new Point(ApplyGridSize(yourNode.Position.X), ApplyGridSize(yourNode.Position.Y));

private double ApplyGridSize(double n)
{
    if (Diagram.Options.GridSize == null)
        return n;

    var gridSize = Diagram.Options.GridSize.Value;
    return gridSize * Math.Floor((n + gridSize / 2.0) / gridSize);
}
```

**Reply by rjtwins** [reply]
Thanks for you quick reply!
